depends/toolchain.cmake.in

```diff
@@ -84,15 +84,19 @@ set(OTOOL "@OTOOL@")
 
 set(CMAKE_FIND_ROOT_PATH "@depends_prefix@")
 set(CMAKE_FIND_ROOT_PATH_MODE_PROGRAM NEVER)
-set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY ONLY)
+set(CMAKE_FIND_ROOT_PATH_MODE_LIBRARY BOTH)
 set(CMAKE_FIND_ROOT_PATH_MODE_INCLUDE ONLY)
 set(CMAKE_FIND_ROOT_PATH_MODE_PACKAGE ONLY)
 set(PKG_CONFIG_PATH "@depends_prefix@/lib/pkgconfig")
 set(PKG_CONFIG_LIBDIR "${PKG_CONFIG_PATH}")
 set(QT_TRANSLATIONS_DIR "@depends_prefix@/translations")
 
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NOT CMAKE_HOST_APPLE)
+  set(CMAKE_FRAMEWORK_PATH "@OSX_SDK@/System/Library/Frameworks")
```

Comment:
Can't remember the exact comments from yesterday, but I think this could use some docs to explain why the SDK is leaking out of the compiler invocation.

Reply:
I've added an explanation comment: https://github.com/hebasto/bitcoin/blob/b0a1bae9f81ff59ca63915b293e9202a9609a4e8/depends/toolchain.cmake.in#L95-L99

Reply:
I'm still confused by this. Does this relate to the previous commit? If I unset `CMAKE_FRAMEWORK_PATH` and drop the previous commit, it still configures, but then fails to link bitcoin-qt.

Reply:
> I'm still confused by this.

What exactly is the source of your confusion: the code or the comment?

Reply:
The code, but I guess I should just look at what cmake does here.

Reply:
https://cmake.org/cmake/help/latest/command/find_library.html describes the role of the `CMAKE_FRAMEWORK_PATH` variable.

Reply:
For Qt's part, please see https://github.com/qt/qtbase/blob/da6e958319e95fe564d3b30c931492dd666bfaff/mkspecs/features/data/cmake/Qt5BasicConfig.cmake.in#L77-L134

Reply:
> Does this relate to the previous commit? If I unset `CMAKE_FRAMEWORK_PATH` and drop the previous commit, it still configures, but then fails to link bitcoin-qt.

No, it doesn't. The `Revert "build, qt: Do not install *.prl files"` commit does not actually made the "prl" files available, which is required for a proper configuration. This is done in the `cmake: Build `bitcoin-qt` executable` commit. The first commit is supposed to make further changes more meaningful and clear.

Unsetting `CMAKE_FRAMEWORK_PATH` would result in a build failure regardless of the commit.